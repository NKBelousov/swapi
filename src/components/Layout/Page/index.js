import React, { Component } from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

const Content = styled.div`
  border-radius: 5px;
  box-shadow: 0 0 5px #000;
  margin: 1rem 0 0 0;
  padding: 1rem;
`;

class Page extends Component {
  static get propTypes() {
    return {
      children: PropTypes.node,
    };
  }
  constructor(props) {
    super(props);
  }
  render() {
    return <Content>{this.props.children}</Content>;
  }
}

export default Page;
