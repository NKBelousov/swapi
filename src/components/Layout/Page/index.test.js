import React from "react";
import renderer from "react-test-renderer";

import Page from "./index.js";

test("Should render correctly", () => {
  const tree = renderer
    .create(
      <Page>
        <h1>Header</h1>
        <h2>Subheader</h2>
      </Page>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
