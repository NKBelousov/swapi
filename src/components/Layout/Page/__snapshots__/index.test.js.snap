// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="sc-bdVaJa gfJEPV"
>
  <h1>
    Header
  </h1>
  <h2>
    Subheader
  </h2>
</div>
`;
